### ISSUE TITLE
Google Play Music shows an incomplete queue

### ISSUE THREAD

**Issue by fenryxo** [tiliado/nuvolaplayer#106]
When the queue button is clicked, the expected pop over is not shown. Notice the presence of a white line and triangle where the queue window should be.

![118ff948-0ea5-11e5-8912-c505aa9e81b1](https://cloud.githubusercontent.com/assets/853706/8083393/7d52cc16-0f84-11e5-9b9c-d2a64b2d032e.png)

**Comment by mathus13** [comment]
Ubuntu 15.04 x86_64 running Unity
Nuvola 3 Beta 
    version: 3.0.0+201506062028.00f51f7
    revision: 00f51f7bac10b2e79bdc
Google Play Music 5.3

I have the same issue. The triangle and white line appear to be the bottom of the queue window, and do shade open and closed as expected when clicking the queue button.

**Comment by dvapelnik** [comment]
Ubuntu 14.04 x86_64 running Unity
Nuvola 3 Beta
Version: 3.0.0+201506062028.00f51f7
Revision: 00f51f7bac10b2e79bdc
![laptop_20150610__17 17 34](https://cloud.githubusercontent.com/assets/1562244/8084613/9e5d6654-0f94-11e5-98be-90d8f3e3d34e.png)

**Comment by RCheesley** [comment]
Ubuntu 14.04  x86_64 running KDE (Kubuntu)
Nuvola Player 3 Beta 3.0.0+201506062028.00f51f7
Revision 00f51f7bac10b2e79bdcf272a35771a30301d0ef

Installed from Tiliado repositories - same issues as reported above, white symbol bottom right and no queue showing.

![google-play-queue](https://cloud.githubusercontent.com/assets/2930593/8089494/235a5bf4-0fa0-11e5-93b6-9410a7f94aa4.png)

**Comment by eigoor** [comment]
Archlinux running Gnome Shell
Version 3.0.0+2015.06.09.0d482d6-1 (installed today from AUR https://aur.archlinux.org/packages/nuvolaplayer-git/)

Working as intended here. Queue is showing were it's supposed to be.
![bildschirmfoto von 2015-06-11 09-49-06](https://cloud.githubusercontent.com/assets/12841918/8102351/b04c7810-101f-11e5-9afc-472b88605d31.png)

**Comment by fenryxo** [comment]
Hello @mathus13, @dvapelnik and @RCheesley. I also need [a debugging output from terminal](https://github.com/tiliado/nuvolaplayer/wiki/Bug-Reporting-Guidelines#debugging-output) to investigate this issue further. Thanks.

**Comment by mathus13** [comment]
here is the output from launch to spawning a radio queue, which opens the queue window:
```
[Master:DEBUG      Gtk] Connecting to session manager
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app KEXP Live Stream at /usr/share/nuvolaplayer3/web_apps/kexp, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Test at /usr/share/nuvolaplayer3/web_apps/test, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Bandcamp at /usr/share/nuvolaplayer3/web_apps/bandcamp, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Deezer at /usr/share/nuvolaplayer3/web_apps/deezer, version 2.3
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Play Music at /usr/share/nuvolaplayer3/web_apps/google_play_music, version 5.3
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Jango at /usr/share/nuvolaplayer3/web_apps/jango, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Mixcloud at /usr/share/nuvolaplayer3/web_apps/mixcloud, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Rdio at /usr/share/nuvolaplayer3/web_apps/rdio, version 4.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Spotify at /usr/share/nuvolaplayer3/web_apps/spotify, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app This is My Jam at /usr/share/nuvolaplayer3/web_apps/this_is_my_jam, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Calendar at /usr/share/nuvolaplayer3/web_apps/google_calendar, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app KEXP Live Stream at /usr/share/nuvolaplayer3/web_apps/kexp, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Test at /usr/share/nuvolaplayer3/web_apps/test, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Bandcamp at /usr/share/nuvolaplayer3/web_apps/bandcamp, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Deezer at /usr/share/nuvolaplayer3/web_apps/deezer, version 2.3
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Play Music at /usr/share/nuvolaplayer3/web_apps/google_play_music, version 5.3
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Jango at /usr/share/nuvolaplayer3/web_apps/jango, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Mixcloud at /usr/share/nuvolaplayer3/web_apps/mixcloud, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Rdio at /usr/share/nuvolaplayer3/web_apps/rdio, version 4.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Spotify at /usr/share/nuvolaplayer3/web_apps/spotify, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app This is My Jam at /usr/share/nuvolaplayer3/web_apps/this_is_my_jam, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Calendar at /usr/share/nuvolaplayer3/web_apps/google_calendar, version 1.0
[Master:DEBUG    DioriteGtk] application.vala:228: XFCE session vanished: org.xfce.SessionManager
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app KEXP Live Stream at /usr/share/nuvolaplayer3/web_apps/kexp, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Test at /usr/share/nuvolaplayer3/web_apps/test, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Bandcamp at /usr/share/nuvolaplayer3/web_apps/bandcamp, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Deezer at /usr/share/nuvolaplayer3/web_apps/deezer, version 2.3
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Play Music at /usr/share/nuvolaplayer3/web_apps/google_play_music, version 5.3
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Jango at /usr/share/nuvolaplayer3/web_apps/jango, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Mixcloud at /usr/share/nuvolaplayer3/web_apps/mixcloud, version 1.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Rdio at /usr/share/nuvolaplayer3/web_apps/rdio, version 4.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Spotify at /usr/share/nuvolaplayer3/web_apps/spotify, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app This is My Jam at /usr/share/nuvolaplayer3/web_apps/this_is_my_jam, version 2.0
[Master:DEBUG    Nuvola] WebAppRegistry.vala:169: Found web app Google Calendar at /usr/share/nuvolaplayer3/web_apps/google_calendar, version 1.0
[Master:INFO     Nuvola] WebAppRegistry.vala:89: Using web app Google Play Music, version 5.3, data dir /usr/share/nuvolaplayer3/web_apps/google_play_music
[Master:DEBUG    Nuvola] MasterController.vala:380: Launch app runner for 'google_play_music': /usr/lib/nuvolaplayer3/apprunner -D -a /usr/share/nuvolaplayer3/web_apps/google_play_music 
[Runner:DEBUG      Gtk] Connecting to session manager
[Master:DEBUG    Nuvola] MasterController.vala:226: Connected to runner server for 'google_play_music'.
[Runner:DEBUG    Nuvola] WebEngine.vala:62: Nuvola WebKit Extension directory: /usr/lib/nuvolaplayer3
[Runner:DEBUG    Nuvola] WebEngine.vala:152: Network Proxy: system settings
[Runner:DEBUG    DioriteGtk] application.vala:228: XFCE session vanished: org.xfce.SessionManager
[Runner:DEBUG    Nuvola] Notifications.vala:175: Notifications: persistence false, actions false, icons false
[Master:DEBUG    Nuvola] MediaKeys.vala:99: GNOME settings daemon appeared: org.gnome.SettingsDaemon, :1.14
[Runner:DEBUG    Nuvola] MPRISProvider.vala:73: Bus acquired: org.mpris.MediaPlayer2.NuvolaPlayer3GooglePlayMusic, registering objects
[Runner:DEBUG    Nuvola] MPRISProvider.vala:90: Bus name acquired: org.mpris.MediaPlayer2.NuvolaPlayer3GooglePlayMusic
[Runner:DEBUG    Nuvola] FormatSupport.vala:88: Trying to play /usr/share/nuvolaplayer3/audio/audiotest.mp3.
[Runner:DEBUG    Nuvola] FormatSupport.vala:88: Pipeline state changed from NULL to READY.
[Runner:DEBUG    Nuvola] FormatSupport.vala:88: Pipeline state changed from READY to PAUSED.
[Runner:DEBUG    Nuvola] FormatSupport.vala:88: Pipeline state changed from PAUSED to PLAYING.
[Runner:DEBUG    Nuvola] WebAppWindow.vala:165: Sidebar position: 1174
[Runner:DEBUG    Nuvola] FormatSupport.vala:88: End of stream for file /usr/share/nuvolaplayer3/audio/audiotest.mp3.
[Runner:DEBUG    Nuvola] FormatSupportCheck.vala:124: Nuvola.WebPlugin: iTunes Application Detector (enabled, not flash) at /usr/lib/mozilla/plugins/librhythmbox-itms-detection-plugin.so: This plug-in detects the presence of iTunes when opening iTunes Store URLs in a web page with Firefox.
[Runner:DEBUG    Nuvola] FormatSupportCheck.vala:124: Nuvola.WebPlugin: Shockwave Flash (enabled, flash) at /usr/lib/mozilla/plugins/flashplugin-alternative.so: Shockwave Flash 11.2 r202
[Worker:DEBUG    Nuvola] main.vala:232: Page 1 created for (null)
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://play.google.com/music/listen#/album//The+Cult/The+Cult, result = true, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Worker:DEBUG    Nuvola] jsapi.vala:519: Nuvola.log: 'Fixed webcomponents: https://raw.githubusercontent.com/tiliado/nuvola-app-google-play/16627e12bf115fcaa97f4abc581382c2251f0e76/webcomponents.js'
[Runner:INFO     Nuvola] AppRunnerController.vala:236: Logged in as Shawn Barratt (mathus13)
[Worker:DEBUG    Nuvola] main.vala:141: Window object cleared: https://play.google.com/music/listen#/album//The+Cult/The+Cult, 0x32c2300, 0x3243280, 0x7f941c29f9b0
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://fls.doubleclick.net/activityi;src=2542116;type=andro238;cat=googl515;ord=1;num=8871220056898.893?, result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://2542116.fls.doubleclick.net/activityi;src=2542116;type=andro238;cat=googl515;ord=1;num=8871220056898.893, result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
https://play.google.com/music/listen#/album//The+Cult/The+CultJS ERROR: Refused to get unsafe header "Location"
JS ERROR: Blocked a frame with origin "https://2542116.fls.doubleclick.net" from accessing a frame with origin "https://play.google.com". Protocols, domains, and ports must match.
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-pause
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-play
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-go-previous
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-go-next
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://play.google.com/music/services/streamingloadalltracks?u=0&xt=CjUKATASMEFNLVdiWGczTGlvcVJaWDQ4M0lvbGdqbWNWT1hJWTI4NGc6MTQzNDcxNTkxNzMyMg%3D%3D&json=%7B%22tier%22%3A2%2C%22requestCause%22%3A1%2C%22requestType%22%3A1%2C%22sessionId%22%3A%228yw1akpejznu%22%7D&format=jsarray, result = true, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://0.client-channel.google.com/client-channel/client?cfg=%7B%222%22%3A%22music%22%2C%228%22%3Afalse%2C%2213%22%3Afalse%7D&ctype=music&xpc=%7B%22cn%22%3A%22wYfdRq5ILm%22%2C%22tp%22%3Anull%2C%22osh%22%3Anull%2C%22ppu%22%3A%22https%3A%2F%2Fplay.google.com%2Frobots.txt%22%2C%22lpu%22%3A%22https%3A%2F%2F0.client-channel.google.com%2Frobots.txt%22%7D, result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://clients4.google.com/invalidation/lcs/client?xpc=%7B%22cn%22%3A%22Ho2g3S2OCZ%22%2C%22tp%22%3Anull%2C%22osh%22%3Anull%2C%22ppu%22%3A%22https%3A%2F%2Fplay.google.com%2Frobots.txt%22%2C%22lpu%22%3A%22https%3A%2F%2Fclients4.google.com%2Frobots.txt%22%7D, result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://accounts.google.com/o/oauth2/postmessageRelay?parent=https%3A%2F%2Fplay.google.com#rpctoken=625833802&forcesecure=1, result = true, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-play
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] WebEngine.vala:507: Navigation, current window: uri = https://plus.google.com/u/0/_/notifications/frame?sourceid=166&hl=en&origin=https%3A%2F%2Fplay.google.com&uc=1&jsh=m%3B%2F_%2Fscs%2Fabc-static%2F_%2Fjs%2Fk%3Dgapi.gapi.en.6zwg_pEP_2A.O%2Fm%3D__features__%2Fam%3DAAQ%2Frt%3Dj%2Fd%3D1%2Frs%3DAItRSTNy-00ELgURzlKy-V-78pEvv5BpHA#pid=166&rpctoken=224828321&_methods=onError%2ConInfo%2ChideNotificationWidget%2CpostSharedMessage%2Creauth%2CsetNotificationWidgetHeight%2CsetNotificationWidgetSize%2CswitchTo%2CnavigateTo%2CsetNotificationText%2CsetNotificationAnimation%2CgetNotificationText%2C_ready%2C_close%2C_open%2C_resizeMe%2C_renderstart&id=I1_1434715925760&parent=https%3A%2F%2Fplay.google.com&pfname=, result = false, frame = (null), type = WEBKIT_NAVIGATION_TYPE_OTHER
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-play
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-pause
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-go-previous
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: can-go-next
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-location
[Runner:DEBUG    Nuvola] DeveloperSidebar.vala:120: Media player notify: artwork-file
```

**Comment by dvapelnik** [comment]
Here is my dedug output https://gist.github.com/dvapelnik/0b69368ff28a7f60d9bd

**Comment by fenryxo** [comment]
Unfortunately, the debug outputs don't show anything suspicions. I will try to reproduce the issue in Ubuntu 14.04/15.04 inside VirtualBox as I cannot reproduce the issue on my system (Debian jessie).

* Issue present
  + Ubuntu 14.04 (trusty) - libwebkit2gtk-3.0-25 2.4.8-1ubuntu1~ubuntu14.04.1
  + Ubuntu 15.04 (vivid) - libwebkit2gtk-4.0-37 2.6.2+dfsg1-4ubuntu1
* Issue doesn't occur
  + Debian 8 (jessie) - libwebkit2gtk-4.0-37 2.6.2+dfsg1-4
  + Arch Linux - webkit2gtk 2.8.3-1

**Comment by mathus13** [comment]
a complete removal and re-installation has resolved this issue

**Comment by courtjestr** [comment]
A complete removal and re-install did not fix it for me. I am on trusty.

**Comment by MRamiro** [comment]
A little insight on the issue, hoping it could be of any help:

I just managed to make the queue visible by removing both the "layout" and "vertical" attributes (though, on closer inspection, it also works if you remove only one of them) on the "paper-dialog#queue-overlay" element. I'm attaching an screenshot for clarity.

Unfortunately, I'm not familiar with whatever javascript magic that powers the Google Play Music app, neither the one that does it for Nuvola, so I have no idea what causes this behaviour nor how it could be fixed.

I'm running Ubuntu 14.04 64bit, with the same Nuvola version as everyone around here:
Nuvola Player 3 Beta 3.0.0+201506062028.00f51f7

![captura de pantalla de 2015-07-02 12 05 55](https://cloud.githubusercontent.com/assets/3259212/8482956/b69922b0-20b2-11e5-90bd-4b38396f4994.png)

**Comment by fenryxo** [comment]
Hello. Can anyone test a fix from git master? Instructions follow:

1. `mkdir -p ~/.local/share/nuvolaplayer3/web_apps`
2. `cd ~/.local/share/nuvolaplayer3/web_apps`
3. `wget https://github.com/tiliado/nuvola-app-google-play/archive/673c8ae13a819c00215d56eaad7d1f635fb2f08e.tar.gz -O nuvola-app-google-play.tar.gz`
4. `tar -xzf nuvola-app-google-play.tar.gz`
5. `nuvolaplayer3 -D -a google_play_music`

Make sure there is a line like `[Master:INFO     Nuvola] WebAppRegistry.vala:89: Using web app Google Play Music, version 5.3, data dir /home/fenryxo/.local/share/nuvolaplayer3/web_apps/nuvola-app-google-play-673c8ae13a819c00215d56eaad7d1f635fb2f08e` in a terminal output. Thanks.

**Comment by MRamiro** [comment]
Hello, fenryxo.

After following your instructions, the queue seems to be working as intended. System specs following:

>Ubuntu 14.04 (trusty) - libwebkit2gtk-3.0-25 2.4.8-1ubuntu1~ubuntu14.04.1

Will report back if anything weird happens.

Thanks for the help.

**Comment by fenryxo** [comment]
Fixed in Google Play Music script 5.4.
